### PR TITLE
Fetch plants from permapeople

### LINKED ISSUE #102: update permapeople.org data
Seems to be a database superior to practicalplants:

https://permapeople.org/knowledgebase/api-docs.html

Following columns would be very interesting for us:

- Nutrition Demand
 - Edible (true/false)
 - Edible parts (list)
 - Edible uses (e.g. oil)
 - Spacing
 - Germination time
 - Seed planting depth
 - Days to harvest (probably Days till harvest is meant?)
 - When to sow (indoors)
 - When to sow (outdoors)
 - Seed viability

Other columns should be included if possible. Excluded columns should be documented for further investigations.

__*EDIT from Nursultan on 09.04.2023:*__

- [x] timeout between requests

### PR DESCRIPTION
**Pull request by aidnurs**
# Permapeople scratch

Closes #102

# Important note

Since the merging and insertion of the data will be the same for Reinsaat and Permapeople, I'll open a separate PR for it. This PR covers a raw fetch from Permapeople into CSV files

## Basics

- [ ] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildserver is happy.
- [ ] The PR is rebased with current master.

## Checklist

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I fixed all affected decisions
- [ ] I added code comments, logging, and assertions as appropriate
- [ ] I mentioned [every code or binary](/.reuse/dep5) not directly written or done by me in [reuse syntax](https://reuse.software/)

## Review

- [ ] Documentation is conforming to [our Documentation Guidelines](/doc/documentation.md)
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to our Coding Guidelines
- [ ] Code is consistent to [our Design Decisions](/doc/decisions)

### CODE REVIEW

File: scraper/README.md [hunk 1]
```diff
@@ -85,3 +92,59 @@ npm run insert <path-to-csv-file> [family|genus]
...
+1. Install dependencies
+
+```shell
+npm install && mkdir -p data
+```
+
+2. Create .env file from .env.example and fill in the required values
+
+```shell
+cp .env.example .env
```
Comment (Bushuo) [inline scraper/README.md:114]:
```suggestion
cp .env.sample .env
```

File: scraper/README.md [hunk 2]
```diff
@@ -85,3 +92,59 @@ npm run insert <path-to-csv-file> [family|genus]
...
+```
+
+`PERMAPEOPLE_KEY_ID` and `PERMAPEOPLE_SECRET_KEY` mentioned in .env are the key ids and secret keys for the Permapeople API.
```
Comment (Bushuo) [inline scraper/README.md:117]:
```suggestion
`PERMAPEOPLE_KEY_ID` and `PERMAPEOPLE_SECRET_KEY` mentioned in .env are the credentials for the Permapeople API.
```

File: scraper/src/compare_datasets.js [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+function sanitizeColumnNames(jsonArray) {
+  return jsonArray.map((obj) => {
+    const keys = Object.keys(obj);
+    keys.forEach((key) => {
+      let newKey = key;
+      newKey = newKey.toLowerCase();
+      if (newKey.includes('&amp;')) {
+        newKey = newKey.split('&amp;').join('and');
+      }
```
Comment (Bushuo) [inline scraper/src/compare_datasets.js:14]: Does this replace all occurrences of `&amp;` with `and`?
I think in this case `newKey.replaceAll('&amp;', 'and')` would make the intention clearer, also the `if` would not be needed.

File: scraper/src/fetch_permapeople.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  let plants = response.data.plants;
+  let i = 0;
+  while (plants.length > 0) {
+    // if (i == 1) break;
+    console.log(`Fetching page ${i + 1}`);
+    plants.forEach((plant) => {
+      plant = plant.data.reduce((acc, curr) => {
+        acc[curr.key] = curr.value;
+        return acc;
+      }, plant);
```
Comment (Bushuo) [inline scraper/src/fetch_permapeople.js:26]: This flattens the data array of the response out into the plant response object, right?
I think a comment would be nice here and a link to the relevant sections in the API documentation.

**Review by Bushuo — APPROVED** [review]
Nicely done 🚀
I just added a few comments to improve the readability.

**Review by Benjaminimal — COMMENTED** [review]
I saw that they do have information on companions and antagonists.
https://permapeople.org/plants/daucus-carota-sativus-carrot#relationships
But I couldn't find any of it in the CSV.
Is it not included in their plants API?

**Comment by markus2330** [comment]
> Is it not included in their plants API?

It was not our plan to include it in these CSVs/DB, as their relationships don't have the quality we would like to have. We will directly import it to your relations later.

File: scraper/src/compare_datasets.js (hunk 3 above)
Comment (aidnurs) [reply scraper/src/compare_datasets.js:14]: Nice suggestion here! For me personally, a little old `if` is closer to the heart xD However, replaceAll is maybe clearer for others, updated it!

File: scraper/src/fetch_permapeople.js (hunk 4 above)
Comment (aidnurs) [reply scraper/src/fetch_permapeople.js:26]: Yeah, it is coming. Thank you overall for your feedback, great catches indeed!

**Comment by aidnurs** [comment]
Hey team,

thank you for all of the feedback! The PR is finally ready to be merged. I outline it here as in the PR description once again:

This PR covers a raw fetch from Permapeople into CSV files. Since the merging and insertion of the data will be the same for Reinsaat and Permapeople, they will be covered with #217 

Sincerely,
Nursultan

**Review by markus2330 — APPROVED** [review]
Let us merge this for the follow-up PR.

**Pull request merged**